### PR TITLE
Ui select doctor

### PR DESCRIPTION
**Pull request by MarkC33**
fixes [AB#10225](https://einnovationteam-mlcsu.visualstudio.com/34fed099-e327-491e-a91e-4822ff25db86/_workitems/edit/10225)
Updated page for the selection of available doctors for an assessment. Currently uses dummy services to return data and doesn't save any data.

### CODE REVIEW

File: Fmas12d.Ui/src/app/services/doctor-list/doctor-list.service.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export class DoctorListService {
     if (searchTerm === '' || searchTerm.length < 3) {
       return of([]);
     }
-    const options = { params: new HttpParams().set('search', searchTerm) };
+    const options = { params: new HttpParams().set('criteria', searchTerm) };
 
     return this.httpClient.get(
       environment.apiEndpoint + '/doctorsearch', options
```
Comment (neildavies11) [inline Fmas12d.Ui/src/app/services/doctor-list/doctor-list.service.ts]: This needs changing, look at the user/search endpoint

File: Fmas12d.Ui/src/app/components/doctor/doctor-select/doctor-select.component.ts [hunk 2]
```diff
@@ -12,49 +19,177 @@ import { tap, switchMap, catchError } from 'rxjs/operators';
...
+      this.routerService.navigatePrevious();
+    }
   }
 
-  get doctorField() {
-    return this.doctorForm.controls.searchDoctor;
+  FetchAvailableDoctors(maxDistance: number) {
+    this.isAvailableDoctorSearching = true;
+    this.userAvailabilityService.getAvailableDoctors(maxDistance)
+    .subscribe(x => {
```
[the remark this answers is not in the document]
Comment (MarkC33) [reply Fmas12d.Ui/src/app/components/doctor/doctor-select/doctor-select.component.ts]: Function removed from component as page design changed (other doctors are now selected on a different page)

File: Fmas12d.Ui/src/app/services/doctor-list/doctor-list.service.ts (hunk 1 above)
Comment (MarkC33) [reply Fmas12d.Ui/src/app/services/doctor-list/doctor-list.service.ts]: updated

File: Fmas12d.Ui/src/app/services/assessment/assessment.service.ts [hunk 3]
```diff
@@ -27,4 +28,13 @@ export class AssessmentService {
       { headers }
     );
   }
+
+  public getAssessment(assessmentId: number) {
+
+    return this.httpClient.get(
+      `${environment.apiEndpoint}/assessment/view/${assessmentId}`
```
[the remark this answers is not in the document]
Comment (MarkC33) [reply Fmas12d.Ui/src/app/services/assessment/assessment.service.ts]: method removed as the functionality has been moevd to a different page

File: Fmas12d.Ui/src/app/interfaces/available-doctor.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+export interface AvailableDoctor {
+  availabilityDetails?: string;
+  distanceFromAssessment: number;
+  doctorGender?: string;
+  doctorName: string;
+  doctorSpeciality?: string;
+  doctorType?: string;
+  id: number;
+  otherAssessmentDetails?: string;
+  selected?: boolean;
```
[the remark this answers is not in the document]
Comment (MarkC33) [reply Fmas12d.Ui/src/app/interfaces/available-doctor.ts]: updated

**Pull request merged**